cypress/utils/table.js

```diff
@@ -57,6 +57,74 @@ function checkNoMatchingRecs() {
   return checkEmptyState('No matching recommendations found');
 }
 
+/**
+ * A column is sorted and then data and a reference column (which can be different than the
+ * sorting one) are compared to see if order matches
+ *
+ * @param {Object} data - following order from API response and complemented with any
+ * modification needed
+ * @param {*} field - key in the data to sort by, or function
+ * @param {string} label - identifier of the column
+ * @param {string} order - ascending or descending
+ * @param {string} columnField - identifier of the column to be used in the comparison with the data
+ * @param {string} dataField - key in the data to be used in the comparison with the data
+ * @param {integer} nExpectedRows - number of expected rows to be displayed in the table
+ * @param {string} validateURL - string expected as sort value in URL. If not provided, it is not checked
+ */
+function checkSorting(
+  data,
+  sortingField,
+  label,
+  order,
+  columnField,
+  dataField,
+  nExpectedRows,
+  validateURL
+) {
+  // get appropriate locators
+  const col = `td[data-label="${label}"]`;
+  const header = `th[data-label="${label}"]`;
+
+  // check number of rows
+  cy.get(col).should('have.length', nExpectedRows);
+
+  // sort by column and verify URL
+  if (order === 'ascending') {
```

Comment:
Not sure if I get your point.

How would you do the `click()` or `dblclick()/click().click()` if we nest it under `if (validateURL)`?

> You also can wrap the URL check in the validateURL check: if true, then call cy.get(header)..stuff, otherwise do nothing

But I need to call `cy.get(header)...` in all cases (if we validate the url or not because we need to click the header to sort by it)

Reply:
ah, yes, true about double clicks.

still you could shorten the code a bit by
1) making the first click,
2) if-closure that clicks additionally if the sort is descending,
3) if-closure that checks URL params if `validateURL`=true.

this is just a nit pick

Reply:
how would the if-closure would look like? Do you have an example?

Reply:
```suggestion
  cy.get(header).find('button').click();
  // sort by column and verify URL
  if (order === 'descending') {
    cy.get(header).find('button').click();
  }

  if (validateURL) {
    cy.location().then((l) => {
      expect(l.search).to.contain(
        `sort=${
          (order === 'descending' ? '-' : '') + columnName2UrlParam(validateURL)
        }`
      );
    });
  }
```

Reply:
While trying to write you an example, I ran into issues with async behavior of the `window` object, and tests were failing because the `window` object was updated after the assertion. However, this https://stackoverflow.com/a/53058423 really helped me, and I learned some new things from there 😄 the code above should work.. at least it succeded at mine locally 🦓

Reply:
I see. So just separate the logic. Sounds good to me. I did the other approach to nest the `expect` in a `then` clause because I was not aware of the trick `cy.location().then`. Said that, aren't we risking to fall into an issue similar to the one you fixed in https://github.com/RedHatInsights/ocp-advisor-frontend/pull/327?

Reply:
True. Maybe we could enclose `if (validateURL) ...` in the `then` closure that you will chain after the `click` command. But it gets complicated. I would prefer the version that you already have in PR.